src/pages/en/core-concepts/sharing-state.md

```diff
@@ -0,0 +1,664 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: Sharing State
+i18nReady: false
+setup: |
+  import Tabs from '../../../components/tabs/Tabs'
+  import UIFrameworkTabs from '~/components/tabs/UIFrameworkTabs.astro'
+  import LoopingVideo from '~/components/LoopingVideo.astro'
+---
+
+> We have islands, but we need boats. ~ [@bholmesdev](https://twitter.com/BHolmesDev)
+
+When building an Astro website with [islands architecture / partial hydration](/en/core-concepts/partial-hydration/), you may have run into this problem: **I want to share state between my components.**
+
+"Typical" UI frameworks like React, Vue, or Svelte may encourage ["context" providers](https://reactjs.org/docs/context.html) for other components to consume. But when [partially hydrating components](/en/core-concepts/framework-components/#hydrating-interactive-components) within Astro or Markdown, you cannot use these context wrappers 😳
+
+We'll need a different solution to create shared stores your components can read and write from: [**nanostores**](https://github.com/nanostores/nanostores).
+
+## Why nanostores?
+
+The [nanostores](https://github.com/nanostores/nanostores) library allows you to author stores that any component can interact with. We recommend nanostores because:
+- **They're lightweight.** Nanostores ship the bare minimum JS you'll need (less than 1 KB) with zero dependencies.
+- **They're framework-agnostic.** This means sharing state between Preact, Svelte, and Vue should be seemless! Astro is built on flexibility, so we love solutions that offer a similar developer experience no matter your preference.
+
+Still, there are a number of alternatives you can explore. These might include:
+- [Svelte's built-in stores](https://svelte.dev/tutorial/writable-stores)
+- [Sending custom browser events](https://developer.mozilla.org/en-US/docs/Web/Events/Creating_and_triggering_events) between components
+- [Solid signals](https://www.solidjs.com/docs/latest) outside of a component context
+
+:::note
```

Comment:
This is fine to do as a follow-up, but I feel like we should have a `:::faq` component that has `details` and `summary` built-in? These don't semantically feel like a note.

Reply:
Agreed! I'll hold off since this PR adds a lot of tooling already, but that definitely came to my mind.